src/main/java/Duke/Tool/Parser.java

```diff
@@ -0,0 +1,259 @@
+
+        int order = task.size();
+        if (cmd.split(" ").length == 1)  {
+            throw new NoDescriptionException("Delete");
+
+        } else if (!isInteger(cmd.split(" ")[1])) {
+            throw new NoCommandException(cmd);
+
+        } else if (Integer.parseInt(cmd.split(" ")[1]) > order) {
+            throw new DeleteWrongIndexException("Delete");
+
+        } else {
+
+            int num = Integer.parseInt(cmd.split(" ")[1]) - 1;
+            return new Delete(num);
+        }
+
+    }
+
+    /**
+     * The method for Todo, Deadline, Event command task process
+     * @param cmd
+     * @param task
+     * @return Task object
+     * @throws NoDescriptionException
+     * @throws NoTimeException
+     * @throws NoCommandException
+     * @throws WrongTimeFormatException
+     */
+    public static Task addTask(String cmd, TaskList task) throws NoDescriptionException, NoTimeException,
+            NoCommandException, WrongTimeFormatException {
+
+        Parser.Operation instruction = Parser.Operation.valueOf(cmd.toUpperCase().split(" ")[0]);
+        int order = task.size();
+        if (cmd.split(" ").length != 1) {
+            switch (instruction) {
+            case TODO:
+
+                if (cmd.split(" ").length == 1) {
+                    throw new NoDescriptionException(instruction.name());
+
+                } else {
+                    return new Todo(cmd.substring(5), false);
+                }
+
+            case DEADLINE:
+                String subString_deadline = cmd.substring(9);
+                if (subString_deadline.split(" /by ").length == 1) {
+                    throw new NoTimeException(instruction.name());
+
+                } else {
+                    try {
+                        return new Deadline(subString_deadline.split(" /by ")[0],
+                                LocalDateTime.parse(subString_deadline.split(" /by ")[1], timeFormat), false);
+                    } catch (Exception e ) {
+                        throw new WrongTimeFormatException(subString_deadline.split(" /by ")[1]);
+                    }
+
+                }
+
+            case EVENT:
+                String subString_event = cmd.substring(6);
+                if (subString_event.split(" /at ").length == 1) {
+                    throw new NoTimeException(instruction.name());
+
+                } else {
+                    try{
+                        return new Event(subString_event.split(" /at ")[0],
+                                LocalDateTime.parse(subString_event.split(" /at ")[1], timeFormat), false);
+                    } catch (Exception e) {
+                        throw new WrongTimeFormatException(subString_event.split(" /at ")[1]);
+                    }
+
+                }
+
+            default:
+                throw new NoCommandException(instruction.name());
+            }
+        }
+
+        else {
```

Comment:
I feel there are too many empty lines in this chunk of code, which affects readibility as a lot more scrolling is needed. Consider cutting down some of them? I noticed this pattern happening elsewhere, but this part is much more pronounced, so I believe you might be adhering to your own internal convention regarding how many blank lines to leave. Even so, I think breaking this convention is worth it if there are going to be this many empty lines.